File: keepr/Models/Keep.cs
namespace keepr.Models;

public class Keep
{
  public int Id { get; set; }
  public string CreatorId { get; set; }
  public string Name { get; set; }
  public string Description { get; set; }
  public string Img { get; set; }
  public int Views { get; set; }
   public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
  // count of how many times its kept
    public int Kept { get; set; }
    // relationships 

    public Account Creator { get; set; }
}

public class VaultKeepItem : Keep
{
    public int VaultKeepId { get; set; }
}

File: keepr.client/src/models/Keep.js
import { Profile } from "./Account.js"

export class Keep{
constructor(data){
this.id = data.id
this.creatorId = data.creatorId
this.name = data.name
this.description = data.description
this.img = data.img || "https://st2.depositphotos.com/1252248/9434/v/600/depositphotos_94342060-stock-illustration-pop-art-comics-style.jpg";
this.views = data.views
this.kept = data.kept
this.creator = new Profile(data.creator)
this.vaultKeepId = data.vaultKeepId
}

}